src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -396,7 +398,13 @@ def module_is_offloaded(module):
         pipeline_is_sequentially_offloaded = any(
             module_is_sequentially_offloaded(module) for _, module in self.components.items()
         )
-        if pipeline_is_sequentially_offloaded and device and torch.device(device).type == "cuda":
+        pipeline_has_bnb_quant = any(_check_bnb_status(module)[0] for _, module in self.components.items())
+        if (
+            not pipeline_has_bnb_quant
```

Comment:
With are we considering bnb quant here ? With 4-bit model, we can move them to the gpu then move back to the cpu with we want to save gpu memory.

Reply:
Good point. Let me try to instead condition this on 8bit loading.